lib/Sema/CSSimplify.cpp

```diff
@@ -2170,6 +2171,22 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
       increaseScore(SK_SyncInAsync);
   }
 
+  /// Whether to downgrade to a concurrency warning.
+  auto isConcurrencyWarning = [&] {
+    if (contextUsesConcurrencyFeatures(DC) ||
+        Context.LangOpts.isSwiftVersionAtLeast(6))
+      return false;
```

Comment:
Perhaps the language version check should be first, since `contextUsesConcurrencyFeatures()` is way more expensive.

(Also, I notice that this looks like an ad-hoc version of the logic in `SendableCheckContext`, but I don't see a profitable way to unify them more.)

Reply:
`contextUsesConcurrencyFeatures` will subsume the Swift >= 6 check shortly.